src/neptune/internal/backends/hosted_neptune_backend.py

```diff
@@ -1103,3 +1104,15 @@ def _get_all_items(get_portion, step):
             previous_items = get_portion(limit=step, offset=len(items))
             items += previous_items
         return items
+
+    @staticmethod
+    def _add_artifact_version_to_request_params(default_request_params: Dict) -> Dict:
```

Comment:
How about moving it to `hosted_artifact_operations` and making it "global" across all artifacts related requests and making `2` some constant?

Reply:
Do you mean to add `X-Neptune-Artifact-Api-Version` to all requests within `hosted_artifact_operations`? 
Initially, I did it like that, but later notice we used it only with `get_artifact_attribute`.

Are there any plans for that header in the future, that I'm not aware of?

Reply:
> Do you mean to add X-Neptune-Artifact-Api-Version to all requests within hosted_artifact_operations?
Initially, I did it like that, but later notice we used it only with get_artifact_attribute

Yes and you're correct that for now it's only needed for `get_artifact_attribute`. But it's mostly to simplify some future feature decompositions.

> Are there any plans for that header in the future, that I'm not aware of?

No, or at least not yet. We're probably going to change the Artifacts API completely.

Reply:
Done